### PR TITLE
[STORM-975] Storm-Kafka trident topology example

### PR DESCRIPTION
**Pull request by arunmahadevan**
A sample word count trident topology to illustrate the use of Transactional Kafka spout and Kafka bolt. This has the following components

1. KafkaBolt that receives random sentences from RandomSentenceSpout and publishes the sentences to a kafka "test" topic.
2. TransactionalTridentKafkaSpout that consumes sentences from the "test" topic, splits it into words, aggregates and stores the word count in a MemoryMapState.
3. DRPC query that returns the word counts by querying the trident state (MemoryMapState).

### CODE REVIEW

File: examples/storm-starter/pom.xml [hunk 1]
```diff
@@ -96,6 +96,12 @@
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.storm</groupId>
+      <artifactId>storm-kafka</artifactId>
```
Comment (caofangkun) [inline examples/storm-starter/pom.xml]: should move ``` <module>examples/storm-starter</module>``` in [pom.xml](https://github.com/apache/storm/blob/master/pom.xml#L164) after ```<module>external/flux</module>```

Comment (arunmahadevan) [inline examples/storm-starter/pom.xml]: @caofangkun thanks. Fixed the dependencies and changed the order.

**Comment by harshach** [comment]
@arunmahadevan can you please file a jira and send a pr with jira number in it.

**Comment by arunmahadevan** [comment]
@harshach added a new jira issue STORM-975

**Comment by harshach** [comment]
@arunmahadevan can you upmerge this with master

**Comment by arunmahadevan** [comment]
@harshach done

**Comment by harshach** [comment]
+1

File: examples/storm-starter/src/jvm/storm/starter/trident/TridentKafkaWordCount.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
+package storm.starter.trident;
```
Comment (HeartSaVioR) [inline examples/storm-starter/src/jvm/storm/starter/trident/TridentKafkaWordCount.java]: @arunmahadevan Please add Apache License comments here.

**Comment by HeartSaVioR** [comment]
Other things look fine.

File: examples/storm-starter/src/jvm/storm/starter/trident/TridentKafkaWordCount.java (hunk 2 above)
Comment (arunmahadevan) [inline examples/storm-starter/src/jvm/storm/starter/trident/TridentKafkaWordCount.java]: @HeartSaVioR done.

**Comment by HeartSaVioR** [comment]
+1. Tested on local machine.

**Pull request merged**